internal/serv/internal/auth/jwt.go

```diff
@@ -153,7 +157,9 @@ func firebaseKeyFunction(token *jwt.Token) (interface{}, error) {
 		}
 	}
 
+	firebasePublicKeys.lock.RLock()
```

Comment:
This lock won't be released if the `if` statement is skipped this should cause the app to hang. Also the function is getting too big and complex to read with locks and unlocks everywhere you might need to split the updating and checking out into separate smaller functions. Maybe using https://golang.org/pkg/sync/#NewCond might be easier I haven't used it before. Anyhow no rush take your time code around shared data access is always tricky.